### ISSUE TITLE
uac_auth() function broken in 4.4.3

### ISSUE THREAD

**Issue by SipSeb** [kamailio/kamailio#797]
We upgraded a system from 4.4.2 to 4.4.3 last week. After that, the uac_auth() function didn't do what it was supposed to do anymore.

This is the failure route in the config file:

```
failure_route[TRUNKAUTH] {
    if (t_is_canceled()) {
        exit;
    }

    if ($T_reply_code == 401 or $T_reply_code == 407) {
        xlog("L_NOTICE", "[$pr:$si:$sp] [$cs $rm cid: $ci  Status: $rs $rr F:$fU T:$tU R:$rU] Remote asked for authentication");
        uac_auth();
        t_relay();
        exit;
    }
}
```

In Kamailio 4.4.2 this works perfectly. In 4.4.3 when this code triggers, we get the following messages in our log:

```
Sep 23 12:25:41 eustatius /usr/sbin/kamailio[13249]: NOTICE: <script>: [udp:1.2.3.4:5060] [102 INVITE cid: asdasdalsjdalsdaskjdlak  Status: <null> <null> F:11111111 T:00491234567890 R:01234567890] Remote asked for authentication
Sep 23 12:25:41 eustatius /usr/sbin/kamailio[13249]: ERROR: <core> [msg_translator.c:417]: clen_builder(): no message body found (missing crlf?)
Sep 23 12:25:41 eustatius /usr/sbin/kamailio[13249]: ERROR: <core> [msg_translator.c:1963]: build_req_buf_from_sip_req(): Error while adjusting Content-Length
Sep 23 12:25:41 eustatius /usr/sbin/kamailio[13249]: ERROR: tm [t_fwd.c:462]: prepare_new_uac(): could not build request
Sep 23 12:25:41 eustatius /usr/sbin/kamailio[13249]: ERROR: tm [t_fwd.c:1723]: t_forward_nonack(): ERROR: t_forward_nonack: failure to add branches
Sep 23 12:25:41 eustatius /usr/sbin/kamailio[13249]: ERROR: tm [tm.c:1415]: _w_t_relay_to(): ERROR: w_t_relay_to: t_relay_to failed
```

After downgrading to 4.4.2, everything works again.

**Comment by scfcplummy** [comment]
Do you use dialog module to increment Cseq? Could be related to this issue #789 ?

**Comment by miconda** [comment]
Can you try with latest master or 4.4 branches? I tried to reproduce, but didn't get the error message, anyhow I found a double free error which might have side effects.

**Comment by miconda** [comment]
Should be fixed, see #819.
